### PR TITLE
Feature/basket list

### PR DESCRIPTION
**Pull request by katisd**
### Page Basket
- BasketSummaryNav
- BasketFoodList

### ETC.
- Fix UserBottomNav positioning: position sticky => fixed
- Refactor Pages: function => arrow function

### CODE REVIEW

File: src/modules/basket/components/BasketSummaryNav.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      </BasketSummaryNavWrapper>
+      <Modal
+        centered={true}
+        open={isModalOpen}
+        maskClosable={false}
+        closable={false}
+        onOk={handleOk}
+        onCancel={() => {
+          if (modalStatus == "pending") return;
+          setIsModalOpen(false);
```
Comment (JiMeow) [inline src/modules/basket/components/BasketSummaryNav.tsx:73]: Should we disable cancel button when modal status is pending?

File: src/modules/basket/components/BasketSummaryNav.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        {modalStatus == "error" && (
+          <Result
+            status="error"
+            title={
+              <Typography.Title level={4}>
+                {"Something went wrong! Please try again."}
+              </Typography.Title>
+            }
+          />
+        )}
```
Comment (JiMeow) [inline src/modules/basket/components/BasketSummaryNav.tsx:103]: Does we have another way to implement content of modal depend on it status ?

File: src/modules/basket/components/BasketSummaryNav.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    "idle" | "pending" | "success" | "error"
+  >("idle");
+  const {
+    token: { colorPrimary, colorBgBase },
+  } = theme.useToken();
+  const openModal = () => {
+    setModalStatus("idle");
+    setIsModalOpen(true);
+  };
+  const handleOk = () => {
```
Comment (JiMeow) [inline src/modules/basket/components/BasketSummaryNav.tsx:21]: Optional: should we change it name? like `handleModalSubmit`, 'handleSendOrder`

**Review by JiMeow — CHANGES_REQUESTED** [review]
Wait surapus for more suggestions

File: src/modules/basket/components/BasketSummaryNav.tsx (hunk 1 above)
Comment (katisd) [reply src/modules/basket/components/BasketSummaryNav.tsx:97]: ✅Done!

File: src/modules/basket/components/BasketSummaryNav.tsx (hunk 2 above)
Comment (katisd) [reply src/modules/basket/components/BasketSummaryNav.tsx]: ✅Done!

File: src/modules/basket/components/BasketSummaryNav.tsx (hunk 3 above)
Comment (katisd) [reply src/modules/basket/components/BasketSummaryNav.tsx]: ✅Done! Change to handleSubmitOrder

File: src/modules/basket/components/BasketFoodCard.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        <ContentGroup>
+          {order.addons?.map((addon) => (
+            <Typography.Text key={addon.id} type="secondary">
+              {addon.name}
+            </Typography.Text>
+          ))}
+        </ContentGroup>
+        {/* TODO navigate to edit link */}
+        <Link style={{ color: colorPrimary }} href={""} target="_blank">
+          Edit
```
Comment (GolfGrab) [inline src/modules/basket/components/BasketFoodCard.tsx:35]: could it be  < Link > <Typography.Link> text </Typography.Link> < Link > for style it without using inline style

File: src/modules/basket/components/BasketFoodCard.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        </Link>
+      </FoodDetails>
+      <Image
+        src={order.food.imagePath ?? ""}
+        alt={order.food.name}
+        width={100}
+        height={100}
+        style={{ objectFit: "cover", borderRadius: "8px" }}
```
Comment (GolfGrab) [inline src/modules/basket/components/BasketFoodCard.tsx:43]: width & hight in next Image stand for resolution of image 
should use higher resolution then use css to scale size down for good quality image

File: src/modules/basket/components/BasketFoodList.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        dataSource={orders}
+        renderItem={(order) => (
+          <List.Item>
+            <BasketFoodCard order={order} key={order.id} />
+          </List.Item>
+        )}
+        style={{ marginInline: "20px" }}
+      />
+    </>
+  );
```
Comment (GolfGrab) [inline src/modules/basket/components/BasketFoodList.tsx:24]: no need for react fragment

File: src/modules/basket/components/BasketSummaryNav.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+          >
+            <Typography.Text style={{ color: colorPrimary }}>
+              Order
+            </Typography.Text>
+          </OrderButton>
+        </ConfigProvider>
+      </BasketSummaryNavWrapper>
+      <Modal
+        centered={true}
+        open={isModalOpen}
```
Comment (GolfGrab) [inline src/modules/basket/components/BasketSummaryNav.tsx:92]: no need for centered={true} just use centered

**Review by GolfGrab — CHANGES_REQUESTED** [review]

**Review by GolfGrab — APPROVED** [review]
niceeee

File: src/modules/basket/components/BasketSummaryNav.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalAPIStatus, setModalStatus] = useState<APIStatus>("idle");
+  const {
+    token: { colorPrimary, colorBgBase },
+  } = theme.useToken();
+  const openModal = () => {
+    setModalStatus("idle");
+    setIsModalOpen(true);
+  };
+  const mapStatusToModalProps: MapStatusToModalProps = {
```
Comment (JiMeow) [inline src/modules/basket/components/BasketSummaryNav.tsx:28]: Optional: move helper to bottom of file

File: src/modules/basket/components/BasketFoodCard.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+              {addon.name}
+            </Typography.Text>
+          ))}
+        </ContentGroup>
+        {/* TODO navigate to edit link */}
+        <Link href={"#"}>
+          <Typography.Link>Edit</Typography.Link>
+        </Link>
+      </FoodDetails>
+      <Image
```
Comment (JiMeow) [inline src/modules/basket/components/BasketFoodCard.tsx:35]: Optional: change to style component

**Review by JiMeow — APPROVED** [review]

**Comment by JiMeow** [comment]
Good job

**Pull request merged**